tests/ContextIntegrationTests.js

```diff
@@ -0,0 +1,318 @@
+/**
+ * GPII Context Integration Tests
+ *
+ * Copyright 2014 Lucendo Development Ltd.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/kettle/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    kettle = fluid.registerNamespace("kettle"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.test.contextIntegration");
+
+require("../index.js");
+
+gpii.loadTestingSupport();
+
+fluid.defaults("gpii.test.contextIntegration.testCaseHolder", {
+    gradeNames: [ "gpii.test.integration.testCaseHolder.linux" ],
```

Comment:
This is a bit backwards - the grade gpii.test.contextIntegration.testCaseHolder should just contain the extra request component. Then we should have gpii.test.contextIntegration.testCaseHolder.linux that inherits from both of them.